scipy/sparse/sputils.py

```diff
@@ -189,12 +190,20 @@ def isintlike(x):
     """Is x appropriate as an index into a sparse matrix? Returns True
     if it can be cast safely to a machine int.
     """
-    if not isscalarlike(x):
```

Comment:
Probably worth commenting that this is strictly an optimization - this is already covered implicitly by the other branches

Reply:
I added a comment to that effect.